src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1870,7 +1870,7 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                 else
                 {
                     var property = sourceMethod.AssociatedSymbol as SourcePropertySymbolBase;
-                    if ((object)property != null && property.IsAutoPropertyWithGetAccessor) // PROTOTYPE: Just IsAutoProperty, or _isAutoPropertyAccessor?
+                    if ((object)property != null && property.IsAutoProperty && property.GetMethod is not null) // PROTOTYPE: Just IsAutoProperty, or _isAutoPropertyAccessor?
```

Comment:
TThis will need future adjustment for mixed scenario. Here we should be caring about whether the accessor is a semi-colon only accessor or not. In fact, this code path shouldn't be reachable for non-semi-colon only accessor. So an assert will probably be needed.

Reply:
> This will need future adjustment for mixed scenario. Here we should be caring about whether the accessor is a semi-colon only accessor or not. In fact, this code path shouldn't be reachable for non-semi-colon only accessor. So an assert will probably be needed.

A simple in-lining of implementation isn't the kind of refactoring I had in mind